Sources/VLCSettingsSpecifierManager.swift

```diff
@@ -0,0 +1,84 @@
+/*****************************************************************************
+ * VLCSettingsSpecifierManager.swift
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright © 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import UIKit
+
+class VLCSettingsSpecifierManager: NSObject {
+    
+    @objc var specifier: IASKSpecifier?
+    var settingsReader: IASKSettingsReader!
+    var settingsStore: IASKSettingsStore!
+    
+    var items: NSArray {
+        guard let items = specifier?.multipleValues() as NSArray? else {
+            fatalError("VLCSettingsSpecifierManager: No rows provided for \(specifier?.key() ?? "null specifier")")
+        }
+        return items
+    }
+    
+    @objc var selectedIndex: IndexPath {
+        let index: Int
+        if let selectedItem = settingsStore.object(forKey: specifier?.key()) {
+            index = items.index(of: selectedItem)
+        } else if let specifier = specifier {
+            index = items.index(of: specifier.defaultValue())
+        } else {
+            fatalError("VLCSettingsSpecifierManager: No specifier provided")
+        }
+        return IndexPath(row: index, section: 0)
+    }
+    
+    @objc init(settingsReader: IASKSettingsReader, settingsStore: IASKSettingsStore) {
+        super.init()
+        self.settingsReader = settingsReader
+        self.settingsStore = settingsStore
```

Comment:
Shouldn't the properties be initialized before calling `super.init()`? Therefore not needing to implicitly unwrap the properties on top?

Reply:
ah good catch. completely missed this :D